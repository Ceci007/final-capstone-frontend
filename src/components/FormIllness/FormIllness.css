@import "https://fonts.googleapis.com/css2?family=Roboto&display=swap";
@import "https://fonts.googleapis.com/css2?family=Raleway:wght@700&display=swap";

.one-form {
  font-family: 'Roboto', sans-serif;
  font-size: 4.4vw;
  border: 0.5px solid rgb(218, 213, 213);
  background-color: white;
  margin: 0 15px 12px 15px;
  border-radius: 11px;
  flex-direction: column;
  align-items: center;
  display: flex;
  justify-content: center;
  height: 100%;
}

.one-parameter {
  display: flex;
  width: 100%;
  justify-content: flex-end;
  margin-top: 15px;
  margin-bottom: 4px;
}

.one-parameter label {
  margin-right: 20px;
  display: flex;
  align-items: center;
}

.one-parameter input,
.one-parameter textarea {
  border: 1px solid gray;
  width: 50vw;
  border-radius: 5px;
  margin-right: 10px;
  font-family: 'Roboto', sans-serif;
  font-size: 4vw;
  padding: 4px;
  margin-left: 3px;
}

.one-parameter input {
  height: 4.5vh;
}

.one-parameter textarea {
  height: 10vh;
}

.one-parameter input:focus,
.one-parameter textarea:focus {
  background-color: aliceblue;
}

.buttons-form {
  display: flex;
  align-items: center;
  margin-top: 9px;
  margin-bottom: 8px;
}

.buttons-form button {
  border: none;
  border-radius: 5px;
  color: white;
  font-weight: bold;
  margin: 5px;
  height: 100%;
  font-size: 3vw;
  width: 20vw;
  cursor: pointer;
  padding: 10px 2px;
}

.buttons-form button:first-child {
  background-color: rgb(173, 220, 145);
}

.buttons-form button:last-child {
  background-color: red;
}

.new-ill {
  justify-content: center;
  display: flex;
}

.newill div h3 {
  margin: 20px;
  font-size: 6vw;
}