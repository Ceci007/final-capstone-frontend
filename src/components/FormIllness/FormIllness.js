import React from 'react';
import PropTypes from 'prop-types';
import './FormIllness.css';
import { connect } from 'react-redux';
import { updateIll } from '../../actions/illness';

class FormIllness extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      name: '',
      description: '',
    };
  }

  componentDidMount = () => {
    const { actionToPerform, buttonId, illness } = this.props;
    if (actionToPerform === 'Save Changes') {
      const ill = illness.filter(x => x.id.toString() === buttonId);
      this.setState({
        name: ill[0].name,
        description: ill[0].description,
      });
    }
  }

  handleChangeName = e => {
    this.setState({
      name: e.target.value,
    });
  }

  handleChangeDescription = e => {
    this.setState({
      description: e.target.value,
    });
  }

  handleSubmit = (name, description) => {
    const { addIllness } = this.props;
    addIllness(name, description);
  }

  handleUpdate = async id => {
    const { name, description } = this.state;

    const {
      user, updateIll, changeEditForm,
    } = this.props;

    const data = {
      id,
      userId: user.user.id,
      name,
      description,
    };

    await updateIll(data);
    changeEditForm();
  }

  render() {
    const { name, description } = this.state;
    const {
      actionToPerform, illness, buttonId, changeEditForm, changeAddForm,
    } = this.props;
    const ill = illness.filter(x => x.id.toString() === buttonId);
    return (
      <div>
        <h3>
          {actionToPerform}
          {' '}
          Illness
        </h3>
        <form
          className="one-form"
          onSubmit={
           actionToPerform === 'Add'
             ? () => this.handleSubmit(name, description) : () => this.handleUpdate(ill[0].id)
}
        >
          <div className="one-parameter">
            <label htmlFor="name">
              Name:
              <input
                required
                id="name"
                type="text"
                name="name"
                defaultValue={buttonId === '0' ? name : ill[0].name}
                onChange={this.handleChangeName}
              />
            </label>
          </div>
          <div className="one-parameter">
            <label htmlFor="description">
              Description:
              <textarea
                id="description"
                name="description"
                defaultValue={buttonId === '0' ? description : ill[0].description}
                onChange={this.handleChangeDescription}
              />
            </label>

          </div>
          <div className="buttons-form">
            {actionToPerform === 'Add' && <button type="submit">{actionToPerform}</button>}
            {actionToPerform === 'Save Changes' && <button type="submit">Save</button>}
            {actionToPerform === 'Add' && <button type="button" onClick={changeAddForm}>Cancel</button>}
            {actionToPerform === 'Save Changes' && <button type="button" onClick={changeEditForm}>Cancel</button>}
          </div>
        </form>
      </div>
    );
  }
}

FormIllness.propTypes = {
  addIllness: PropTypes.func,
  actionToPerform: PropTypes.string,
  changeAddForm: PropTypes.func,
  illness: PropTypes.instanceOf(Array),
  buttonId: PropTypes.string,
  updateIll: PropTypes.func,
  changeEditForm: PropTypes.func,
  user: PropTypes.shape({
    user: PropTypes.shape({
      id: PropTypes.number,
    }),
  }),

};

FormIllness.defaultProps = {
  changeAddForm: () => {},
  addIllness: () => {},
  actionToPerform: '',
  illness: [],
  buttonId: '0',
  updateIll: () => {},
  changeEditForm: () => {},
  user: {},
};

const mapStateToProps = state => ({
  user: state.user,
  illness: state.illness,
});
const mapDispatchToProps = dispatch => ({
  updateIll: data => dispatch(updateIll(data)),
});

export default connect(mapStateToProps, mapDispatchToProps)(FormIllness);
