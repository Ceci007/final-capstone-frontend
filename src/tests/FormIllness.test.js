import thunk from 'redux-thunk';
import { Provider } from 'react-redux';
import { create } from 'react-test-renderer';
import { createStore, applyMiddleware } from 'redux';
import combinedReducers from '../reducers';
import FormIllness from '../components/FormIllness/FormIllness';

test('FormIllness renders correctly', () => {
  const store = createStore(combinedReducers, applyMiddleware(thunk));
  const tree = create(
    <Provider store={store}>
      <FormIllness />
    </Provider>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
