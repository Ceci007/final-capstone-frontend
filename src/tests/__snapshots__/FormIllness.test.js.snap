// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders correctly 1`] = `
<div>
  <h3>
    
     
    Illness
  </h3>
  <form
    className="one-form"
    onSubmit={[Function]}
  >
    <div
      className="one-parameter"
    >
      <label
        htmlFor="name"
      >
        Name:
        <input
          defaultValue=""
          id="name"
          name="name"
          onChange={[Function]}
          required={true}
          type="text"
        />
      </label>
    </div>
    <div
      className="one-parameter"
    >
      <label
        htmlFor="description"
      >
        Description:
        <textarea
          defaultValue=""
          id="description"
          name="description"
          onChange={[Function]}
        />
      </label>
    </div>
    <div
      className="buttons-form"
    />
  </form>
</div>
`;

exports[`FormIllness renders correctly 1`] = `
<div>
  <h3>
    
     
    Illness
  </h3>
  <form
    className="one-form"
    onSubmit={[Function]}
  >
    <div
      className="one-parameter"
    >
      <label
        htmlFor="name"
      >
        Name:
        <input
          defaultValue=""
          id="name"
          name="name"
          onChange={[Function]}
          required={true}
          type="text"
        />
      </label>
    </div>
    <div
      className="one-parameter"
    >
      <label
        htmlFor="description"
      >
        Description:
        <textarea
          defaultValue=""
          id="description"
          name="description"
          onChange={[Function]}
        />
      </label>
    </div>
    <div
      className="buttons-form"
    />
  </form>
</div>
`;
